src/FactSystem/FactMetaData.cc

```diff
@@ -893,8 +912,15 @@ const FactMetaData::AppSettingsTranslation_s* FactMetaData::_findAppSettingsDist
 {
     for (size_t i=0; i<sizeof(_rgAppSettingsTranslations)/sizeof(_rgAppSettingsTranslations[0]); i++) {
         const AppSettingsTranslation_s* pAppSettingsTranslation = &_rgAppSettingsTranslations[i];
-        if (pAppSettingsTranslation->rawUnits == rawUnits.toLower() &&
-           (pAppSettingsTranslation->unitType == UnitDistance && pAppSettingsTranslation->unitOption == qgcApp()->toolbox()->settingsManager()->unitsSettings()->distanceUnits()->rawValue().toUInt())) {
+
+        if (!(rawUnits == pAppSettingsTranslation->rawUnits) && !(rawUnits.toLower() == pAppSettingsTranslation->rawUnits)) {
```

Comment:
Shouldn't this just be one test which does toLower on both strings? More understandable.

Reply:
Which you can't do because one is a char* not a QString.

Reply:
Maybe just move pAppSettingsTranslation->rawUnits into a QString ahead of time so you can toLower it.